### DISCUSSION TITLE
Database driver plug-in development questions:   <extension ...>

### DISCUSSION THREAD

**Discussion by yangmeng66** [dbeaver/dbeaver#14973] [DBeaver Development]
Everybody is good! I have some questions that I hope can only guide me:

Our company was developing our own database (similar to ClickHouse) and wanted to use your Dbeaver, but also wanted to do something custom, as clickHouse writes here:

```
    <extension point="org.jkiss.dbeaver.generic.meta">
        <meta id="clickhouse" class="org.jkiss.dbeaver.ext.clickhouse.model.ClickhouseMetaModel" driverClass="ru.yandex.clickhouse.ClickHouseDriver" dialect="clickhouse"/>
    </extension>
    <extension point="org.jkiss.dbeaver.sqlDialect">
        <dialect id="clickhouse" parent="generic" class="org.jkiss.dbeaver.ext.clickhouse.model.ClickhouseSQLDialect" label="Clickhouse" description="Yandex Clickhouse SQL dialect." icon="platform:/plugin/org.jkiss.dbeaver.ext.generic/icons/clickhouse_icon.png">
        </dialect>
    </extension>

    <extension point="org.jkiss.dbeaver.objectManager">
        <manager class="org.jkiss.dbeaver.ext.clickhouse.edit.ClickhouseTableManager" objectType="org.jkiss.dbeaver.ext.clickhouse.model.ClickhouseTable"/>
        <manager class="org.jkiss.dbeaver.ext.clickhouse.edit.ClickhouseViewManager" objectType="org.jkiss.dbeaver.ext.clickhouse.model.ClickhouseView"/>
    </extension>
```

But unfortunately I couldn't find the relevant driver docs for "extension..." Thank you very much

**Comment by LonwoLonwo** [comment]
Hello @yangmeng66 

Unfortunately, we do not have suitable documentation about plugins development.
You can copy https://github.com/dbeaver/dbeaver/tree/devel/plugins/org.jkiss.dbeaver.ext.clickhouse entirely and change it for your purposes.
(Like for OceanBase https://github.com/dbeaver/dbeaver/pull/13449)

"extension point" - standard Eclipse annotation so that you will find much information on the Internet :)

**Comment by yangmeng66** [comment]
Ok, thanks for your reply!!
